src/common/Executor.cpp

```diff
@@ -612,40 +611,90 @@ void Executor::setupSuite()
         }
-        tuning_names[vid].resize(tuning_names_order_map.size());
-        for (auto const& tuning_name_idx_pair : tuning_names_order_map) {
-          tuning_names[vid][tuning_name_idx_pair.second] = tuning_name_idx_pair.first;
+      }
+      // Check for invalid tuning input
+      Svector invalid;
+      for (string const& tuning_name: selected_tuning_names) {
+        if (all_tunings.find(tuning_name) == all_tunings.end()) {
+          invalid.push_back(tuning_name);
         }
-        // reorder to put "default" first
-        auto default_order_iter = tuning_names_order_map.find(KernelBase::getDefaultTuningName());
-        if (default_order_iter != tuning_names_order_map.end()) {
-          size_t default_idx = default_order_iter->second;
-          std::string default_name = std::move(tuning_names[vid][default_idx]);
-          tuning_names[vid].erase(tuning_names[vid].begin()+default_idx);
-          tuning_names[vid].emplace(tuning_names[vid].begin(), std::move(default_name));
+      }
+      run_params.setInvalidTuningInput(invalid);
+      // Check for invalid exclude tuning input
+      invalid.clear();
+      for (string const& tuning_name: excluded_tuning_names) {
+        if (all_tunings.find(tuning_name) == all_tunings.end()) {
+          invalid.push_back(tuning_name);
         }
       }
-
-      //
-      // If we've gotten to this point, we have good input to run.
-      //
-      if ( run_params.getInputState() != RunParams::DryRun &&
-           run_params.getInputState() != RunParams::CheckRun ) {
-        run_params.setInputState(RunParams::PerfRun);
+      run_params.setInvalidExcludeTuningInput(invalid);
+      // Validate invalid input
+      if ( !(run_params.getInvalidTuningInput().empty()) ||
+           !(run_params.getInvalidExcludeTuningInput().empty())) {
+        run_params.setInputState(RunParams::BadInput);
       }
 
+      if (run_params.getInputState() != RunParams::BadInput) { // If tunings input is valid
+        for (VariantID vid : variant_ids) {
+          for (const KernelBase* kernel : kernels) {
+            for (std::string const& tuning_name :
+                kernel->getVariantTuningNames(vid)) {
+              if (tuning_names_order_map.find(tuning_name) ==
+                  tuning_names_order_map.end()) {
+                if ((selected_tuning_names.empty() || find(selected_tuning_names.begin(), selected_tuning_names.end(), tuning_name) != selected_tuning_names.end()) // If argument is not provided or name is selected
+                      &&
+                    find(excluded_tuning_names.begin(), excluded_tuning_names.end(), tuning_name) == excluded_tuning_names.end()) { // name does not exist in exclusion list
+                      tuning_names_order_map.emplace(
+                          tuning_name, tuning_names_order_map.size()); // Add tuning name to map
+                }
+              }
+            }
+          }
+
+          std::vector<std::string> final_tunings;
```

Comment:
Should this be in the variants loop?

Reply:
`final_tunings` is in the `for (VariantID vid : variant_ids) {` loop unless I am losing my mind. check again?

Reply:
True, but should it have been? If this is only used for adiak, it can all be in the ifdef.

Reply:
What I'm driving at is this value is added for each variant, but what does that actually do, does it override the value resulting in potentially missing some tunings from other variants?

Reply:
[8c5aa96](https://github.com/LLNL/RAJAPerf/pull/333/commits/8c5aa965084218704b79891d73f9e4cf8ab486c2) Ok I see what you mean. You are correct; it's only for the metadata.

Reply:
> What I'm driving at is this value is added for each variant, but what does that actually do, does it override the value resulting in potentially missing some tunings from other variants?

[4834b0e](https://github.com/LLNL/RAJAPerf/pull/333/commits/4834b0ed80287d9a548ffc534f142e6df33080ce) Moving it out of the loop ensures that we aren't missing anything since `tuning_names` has all the tunings.